File: weknot_android/app/src/main/java/com/example/weknot_android/room/sharedpreference/Token.java
package com.example.weknot_android.room.sharedpreference;

import android.content.Context;
import android.content.ContextWrapper;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.f2prateek.rx.preferences2.Preference;
import com.f2prateek.rx.preferences2.RxSharedPreferences;

import io.reactivex.Observable;
import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;
import io.reactivex.observers.DisposableSingleObserver;
import io.reactivex.schedulers.Schedulers;

public class Token extends ContextWrapper{

    private String token;

    public Token(Context context) {
        super(context);
    }

    public void setToken(String token) {

        SharedPreferences sharedPreferences = getSharedPreferences("weknot",MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        editor.putString("token", token);

        editor.commit();
    }

    public String getToken() {

        SharedPreferences sharedPreferences = getSharedPreferences("weknot",MODE_PRIVATE);
        RxSharedPreferences rxPreferences = RxSharedPreferences.create(sharedPreferences);

        Preference<String> tokenObservable = rxPreferences.getString("token");

        tokenObservable.asObservable().subscribe(token -> this.token = token);

        return token;

    }
}

File: weknot_android/app/src/main/java/com/example/weknot_android/database/sharedpreference/Token.kt
package com.example.weknot_android.database.sharedpreference

import android.annotation.SuppressLint
import android.content.Context
import android.content.ContextWrapper
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import com.f2prateek.rx.preferences2.Preference
import com.f2prateek.rx.preferences2.RxSharedPreferences

class Token(context: Context) : ContextWrapper(context) {
    var token: String = ""
        @SuppressLint("CheckResult")
        get() {
            val sharedPreferences: SharedPreferences = getSharedPreferences("weknot", Context.MODE_PRIVATE)
            val rxPreferences = RxSharedPreferences.create(sharedPreferences)
            val tokenObservable: Preference<String> = rxPreferences.getString("token")
            tokenObservable.asObservable().subscribe { token: String -> field = token }
            return field
        }
        @SuppressLint("ApplySharedPref")
        set(token) {
            val sharedPreferences: SharedPreferences = getSharedPreferences("weknot", Context.MODE_PRIVATE)
            val editor: Editor = sharedPreferences.edit()
            editor.putString("token", token)
            editor.commit()
        }

}